spec/locale.html

```diff
@@ -377,10 +391,9 @@ <h1>
       <dl class="header">
       </dl>
       <emu-alg>
-        1. Assert: _locale_ can be matched by the <code>unicode_locale_id</code> Unicode locale nonterminal.
-        1. Let _languageId_ be the longest prefix of _locale_ matched by the <code>unicode_language_id</code> Unicode locale nonterminal.
-        1. Assert: The first subtag of _languageId_ can be matched by the <code>unicode_language_subtag</code> Unicode locale nonterminal.
-        1. Return the first subtag of _languageId_.
+        1. Let _baseName_ be GetLocaleBaseName(_locale_).
```

Comment:
Is there a reason why the calls to `GetLocaleBaseName` in `GetLocaleLanguage`, `GetLocaleRegion`, etc. can't occur in the associated getters rather than in the AOs themselves, in the interest of not calling `GetLocaleBaseName` twice when calling these AOs from `UpdateLanguageId`?